### PR TITLE
[NU-1609] fixed: Nu runtime memory leak 

### PR DESCRIPTION
**Pull request by mk-software-pl**
## Describe your changes

`SynchronousExecutionContextAndIORuntime` provides sync IORuntime. The runtime is rather dummy and there is no simple way to close it properly. We thought that there was no simple way to close it, but the closing is not required because it's a dummy. It figures out that it's not so true, because by default, during creation, MBean is registered. We don't need the monitoring functionality provided by the MBean, so we simply disable it before creating the sync IORuntime. Now, we consider it really a dummy one.

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Review by raphaelsolarski — APPROVED** [review]
Epic!

File: docs/Changelog.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 * [#5843](https://github.com/TouK/nussknacker/pull/5843) Added new endpoint to provide statistics URL
 * [#5873](https://github.com/TouK/nussknacker/pull/5873) Handle list of the anonymous statistic URLs, and send them in the interval
 * [#5889](https://github.com/TouK/nussknacker/pull/5889) Decision Table component parameters validation improvement
+* [#5898](https://github.com/TouK/nussknacker/pull/5898) fixed Nu runtime memory leak
```
Comment (lciolecki) [inline docs/Changelog.md:19]: Fixed

File: utils/utils/src/main/scala/pl/touk/nussknacker/engine/util/SynchronousExecutionContextAndIORuntime.scala [hunk 2]
```diff
@@ -49,4 +51,22 @@ object SynchronousExecutionContextAndIORuntime extends LazyLogging {
     override def monotonicNanos(): Long = System.nanoTime()
   }
 
+  // note: even if the provided by us IORuntime is dummy, it registers MBean for a sake of monitoring fibers. We don't
+  //       need it, so we just disable it. There is no other way to do it by setting system property. After IORuntime
+  //       creation we clear the property (to not affect creating other /not existing currently/ IORuntimes). Note that
+  //       we change here the global state, so this is not a bullet proof solution.
+  private def withDisableCatsTracingMode(create: => IORuntime): IORuntime = synchronized {
+    val tracingModePropertyName = "cats.effect.tracing.mode"
```
Comment (lciolecki) [inline utils/utils/src/main/scala/pl/touk/nussknacker/engine/util/SynchronousExecutionContextAndIORuntime.scala:59]: Can we declare this property at the beginning `SynchronousExecutionContextAndIORuntime`:
```
private val TracingModePropertyName = "cats.effect.tracing.mode"
```
?

File: docs/Changelog.md (hunk 1 above)
Comment (mk-software-pl) [reply docs/Changelog.md:19]: we will fix this later

File: utils/utils/src/main/scala/pl/touk/nussknacker/engine/util/SynchronousExecutionContextAndIORuntime.scala (hunk 2 above)
Comment (mk-software-pl) [reply utils/utils/src/main/scala/pl/touk/nussknacker/engine/util/SynchronousExecutionContextAndIORuntime.scala:59]: nope, IMO the current place of declaration is optimal

**Pull request merged**